packages/myst-to-tex/src/math.ts

```diff
@@ -1,5 +1,35 @@
 import type { Handler, ITexSerializer } from './types.js';
 
+// Top level environments in amsmath version 2.1 (and eqnarray), see:
+// http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/required/amsmath/amsldoc.pdf
+const ENVIRONMENTS = [
+  'equation',
+  'multline',
+  'gather',
+  'align',
+  'alignat',
+  'flalign',
+  'matrix',
+  'pmatrix',
+  'bmatrix',
+  'Bmatrix',
+  'vmatrix',
+  'Vmatrix',
+  'eqnarray',
+];
+
+const RE_OPEN = new RegExp(`^\\\\begin{(${ENVIRONMENTS.join('|')})([*]?)}`);
+
+function isAmsmathEnvironment(value: string) {
+  const matchOpen = value.trim().match(RE_OPEN);
+  if (!matchOpen) return false;
+  const [, environment, numbered] = matchOpen;
+  const end = `\\end{${environment}${numbered}}`;
+  const matchClose = value.trim().endsWith(end);
+  if (!matchClose) return false;
```

Comment:
Should it throw an execption here? The LaTeX part would also fail to compile anyway, and at least it can display nicer error message than the LaTeX logs

Reply:
There will already be an error shown when katex tries to compile this, so I think it is ok to leave this out here.